### ISSUE TITLE
Compile issues in latest release candidate

### ISSUE THREAD

**Issue by oubiwann** [lfe/rebar3_lfe#69]
* Errors when copying `app.src` files
* Issues with copying include files (sometimes have to compile projects multiple times before includes are finally copied ...)

**Comment by oubiwann** [comment]
Notes from @rvirding:
```
Error: undef
[{rebar_app_utils,app_src_to_app,
                  ["/Users/rv/sune/bert/_build/default/lib/bert",
                   "/Users/rv/sune/bert/apps/bert/src/bert.app.src"],
                  []},
 {rebar3_lfe_utils,copy_app_src,1,
                   [{file,"/Users/rv/sune/bert/_build/default/plugins/rebar3_lfe/src/rebar3_lfe_utils.erl"},
                    {line,35}]},
 {rebar3_lfe_prv_compile,compile_app,1,
                         [{file,"/Users/rv/sune/bert/_build/default/plugins/rebar3_lfe/src/rebar3_lfe_prv_compile.erl"},
                          {line,59}]},
...
```

Follow-up comments from @rvirding:
> It looks like there is a `rebar_app_utils:app_src_to_app/3` function while `rebar3_lfe_utils:copy_app_src/1` is calling a `/2` function.  I checked the code I have in `.cache/rebar3/plugins/rebar3_lfe/src/rebar3_lfe_utils.erl` line 35 and it does definitely call `/2`.

**Comment by oubiwann** [comment]
I think the error that Robert is seeing has been addressed with the commit that introduced this fix in `src/rebar3_lfe_utils.erl`:
```
app_file(OutDir, SrcFile, State) ->
    %% Rebar screwed us over, here: they changed the arity of
    %% rebar_app_utils:app_src_to_app between their 3.15 and 3.16 release,
    %% so we have to do crazy here:
    UtilsExports = proplists:get_value(exports, rebar_app_utils:module_info()),
    App2AppArity = proplists:get_value(app_src_to_app, UtilsExports),
    case App2AppArity of
        2 -> rebar_app_utils:app_src_to_app(OutDir, SrcFile);
        3 -> rebar_app_utils:app_src_to_app(OutDir, SrcFile, State);
        A -> {error, io_lib:format("Unexpected arity: rebar_app_utils:app_src_to_app/~p", [A])}
    end.
```

**Comment by oubiwann** [comment]
The error I continue to see after that fix is this one, in the rebar3 output of a project's build:
```
===> 		Failed to copy "/Users/oubiwann/lab/lfe/loise/src/loise.app.src": enoent
```

This is odd because:
1. the file exists, and
2. the copy is actually successful 

```
$ ls -al src/loise.app.src
-rw-r--r--  1 oubiwann  staff  1754 Jul 20 16:18 src/loise.app.src
```
```
$ ls -al _build/default/lib/loise/ebin/loise.app
-rw-r--r--  1 oubiwann  staff  1055 Oct 31 12:23 _build/default/lib/loise/ebin/loise.app
```

**Comment by oubiwann** [comment]
Okay, I've got a fix in place for this, done in 5141374.

And 0.4.0 is published here: https://hex.pm/packages/rebar3_lfe/0.4.0

**Comment by oubiwann** [comment]
Tested new version with an old project and all the errors I'd seen previously are now gone 😅 

Closing ...
